spec/controllers/miq_report_controller/reports/editor_spec.rb

```diff
@@ -174,6 +174,47 @@
         end
       end
     end
+
+    def non_empty_category
+      cat = FactoryGirl.create(:classification, :name => "non_empty", :description => "Has entries")
+      cat.add_entry(:name => "foo", :description => "Foo")
+      cat.add_entry(:name => "bar", :description => "Bar")
+      cat
+    end
+
+    def empty_category
+      FactoryGirl.create(:classification, :name => "empty", :description => "Zero entries")
+    end
+
+    describe "#categories_hash" do
+      it "returns non-empty categories" do
+        non_empty_category
+        empty_category
+        expect(controller.send(:categories_hash)).to eq("non_empty" => "Has entries")
+      end
+
+      it "omits show: false categories" do
+        non_empty_category.update(:show => false)
```

Comment:
I don't know what you mean here. You can change the parameters of a single let within another context to only have one write, and it looks pretty spiffy. I can show you an example sometime if you like. 

> Made the overrides params of the helper passed into the factory, writing once per example that calls it. PTAL.

This is fine, just go with this.